File: pharmacy-v1/pharmacy-backend-microservice/product-service/src/main/java/com/pharmacy_backend/product_service/service/WishlistService.java
package com.pharmacy_backend.product_service.service;


import com.pharmacy_backend.common.dto.response.ApiResponse;
import com.pharmacy_backend.product_service.dto.response.PageResponse;
import com.pharmacy_backend.product_service.dto.response.ProductResponse;

import java.util.List;

public interface WishlistService {
    ApiResponse<PageResponse<List<ProductResponse>>> getMyWishlist(int pageIndex, int pageSize);
    ApiResponse<Void> addProductToWishlist(Long productId);
    ApiResponse<Void> removeProductsFromWishlist(List<Long> productIds);
    ApiResponse<Void> clearWishlist();
}

File: pharmacy-v1/fe/pharmacy-app/src/api/wishlistService.ts
import axiosClient from './axiosClient';
import { ApiResponse, PageResponse } from '../types';
import { Product } from '../types/product';

const wishlistService = {
  // 1. Lấy danh sách yêu thích (có phân trang)
  getMyWishlist: (pageIndex: number = 1, pageSize: number = 10) => {
    return axiosClient.get<ApiResponse<PageResponse<Product>>>('/wishlists/my-wishlist', {
      params: { pageIndex, pageSize },
    });
  },

  // 2. Thêm sản phẩm vào danh sách yêu thích
  addToWishlist: (productId: number) => {
    return axiosClient.post<ApiResponse<void>>('/wishlists', null, {
      params: { productId },
    });
  },

  // 3. Xóa một hoặc nhiều sản phẩm khỏi danh sách yêu thích
  removeFromWishlist: (productIds: number[]) => {
    return axiosClient.delete<ApiResponse<void>>('/wishlists/remove', {
      params: { productIds },
      paramsSerializer: (params) => {
        const searchParams = new URLSearchParams();
        (params.productIds as number[]).forEach((id) =>
          searchParams.append('productIds', id.toString())
        );
        return searchParams.toString();
      },
    });
  },

  // 4. Xóa toàn bộ danh sách yêu thích
  clearWishlist: () => {
    return axiosClient.delete<ApiResponse<void>>('/wishlists/clear');
  },
};

export default wishlistService;
